.github/workflows/weekly_vuln_scan.yml

```diff
@@ -1,41 +1,60 @@
+# SPDX-FileCopyrightText: 2024 PNED G.I.E.
+# SPDX-License-Identifier: Apache-2.0
+
 name: Scan Latest Two Semantically Versioned Docker Image Tags with ORT
 
 on:
   schedule:
-    - cron: "0 14 * * 1" # Every Monday at 14:00 UTC
+    - cron: "0 14 * * 1" # Works on Every Monday at 14:00 UTC
   workflow_dispatch:
 
 jobs:
   fetch-and-scan:
     runs-on: ubuntu-latest
     steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
       - name: Setup Docker
         run: |
           sudo service docker start
+          docker ps
 
       - name: Login to GitHub Container Registry
         run: |
           echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
+          docker info
 
       - name: List Docker Image Tags and Fetch Latest Two
         run: |
           IMAGE="ghcr.io/genomicdatainfrastructure/gdi-userportal-frontend"
-          TAGS=$(curl -s -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" "https://ghcr.io/v2/${IMAGE}/tags/list" | jq -r '.tags[]' | sort -Vr)
-          echo "Tags fetched: $TAGS"
-          # Filtering for the last patch of each of the latest two minor versions
-          LATEST_PATCHES=$(echo $TAGS | tr ' ' '\n' | grep -E '^v[0-9]+\.[0-9]+\.[0-9]+$' | sort -Vr | awk -F '.' '{key=$1"."$2; if(!a[key] || $3 > a[key]) {a[key]=$0}} END {for (i in a) print a[i]}' | head -n 2)
-          echo "Latest two patches: $LATEST_PATCHES"
-          echo "LATEST_PATCHES=$LATEST_PATCHES" >> $GITHUB_ENV
+          echo "Fetching Docker image tags..."
+          RESPONSE=$(curl -s -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" "https://api.github.com/orgs/genomicdatainfrastructure/packages/container/gdi-userportal-frontend/versions")
+
+          # Log the response for debugging
+          echo "Response from curl: $RESPONSE"
+
+          # Extract tags and creation dates, then sort by creation date
+          TAGS_WITH_DATES=$(echo "$RESPONSE" | jq -r '.[] | .metadata.container.tags[] as $tag | "\($tag) \(.created_at)"' | sort -k2 -r)
+
+          # Filter out the 'main' tag and get the latest two tags
+          LATEST_TAGS=$(echo "$TAGS_WITH_DATES" | awk '$1 != "main" {print $1}' | head -n 2 | tr '\n' ' ')
+
+          # Print the results
+          echo "Tags fetched: $LATEST_TAGS"
```

Comment:
question: why did you need to fix this algorithm?

We need to check only the latest from the current version and the latest from the previous version. 

Fetching the last two sorted by created date will not fulfil this requirement, as it may fetch the last 2 of the same version.